src/reducer-immutable.js

```diff
@@ -0,0 +1,40 @@
+
+import warning from 'warning';
+
+import createReducer from './utils/createReducer';
+import {
+  SHOW,
+  REPLACE,
+  DISMISS,
+} from './actions';
+
+let reducer; // eslint-disable-line import/no-mutable-exports
+
+try {
+  const fromJS = require('immutable').fromJS; // eslint-disable-line global-require
+
+  const initialState = fromJS({
+    show: false,
+    title: '',
+  });
+
+  reducer = createReducer(initialState, {
```

Comment:
I am not familiar with immutablejs, but I thought it should be:

```js
reducer = createReducer(initialState, {
  [SHOW]: (state, { payload }) => fromJS({
    show: true,
    ...payload,
  }),

  [REPLACE]: (state, { payload }) => fromJS({
    ...state,
    ...payload,
  }),

  [DISMISS]: () => initialState,
});
```

Correct me if I am wrong.